Calendars/Calendars.Plugin.Tizen/CalendarsImplementation.cs

```diff
@@ -0,0 +1,490 @@
+		public Task<Calendar> GetCalendarByIdAsync(string externalId)
+		{
+			CalendarRecord calRecord = null;
+			Calendar calendar = null;
+
+			try
+			{
+				calRecord = new CalendarRecord(Book.Uri);
+				calRecord.Set<string>(Book.Name, "org.tizen.calendar");
+				calendar = new Calendar();
+				calendar.Name = calRecord.Get<string>(Book.Name);
+				calendar.ExternalID = calRecord.Get<int>(Book.Id).ToString();
+				calendar.CanEditEvents = true;
+				calendar.CanEditCalendar = false;
+				calendar.AccountName = calRecord.Get<int>(Book.AccountId).ToString();
+				calendar.Color = calRecord.Get<string>(Book.Color);
+			}
+			catch (Exception ex)
+			{
+				throw new Exception(ex.Message);
+			}
+			finally
+			{
+				calRecord?.Dispose();
+				calRecord = null;
+			}
+
+			return Task.Run(() => calendar);
+		}
+
+		/// <summary>
+		/// Gets all events for a calendar within the specified time range.
+		/// </summary>
+		/// <param name="calendar">Calendar containing events</param>
+		/// <param name="start">Start of event range</param>
+		/// <param name="end">End of event range</param>
+		/// <returns>Calendar events</returns>
+		/// <exception cref="System.ArgumentException">Calendar does not exist on device</exception>
+		/// <exception cref="System.UnauthorizedAccessException">Calendar access denied</exception>
+		/// <exception cref="Plugin.Calendars.Abstractions.PlatformException">Unexpected platform-specific error</exception>
+		public Task<IList<CalendarEvent>> GetEventsAsync(Calendar calendar, DateTime start, DateTime end)
+		{
+			return Task.Run<IList<CalendarEvent>>(() =>
+			{
+				var calendars = new List<CalendarEvent>();
+
+				CalendarManager calManager = null;
+				CalendarList calList = null;
+				CalendarRecord calRecord = null;
+
+				if (calList.Count == 0)
+					return calendars;
+
+				calList.MoveFirst();
+
+				try
+				{
+					do
+					{
+						calManager = new CalendarManager();
+						calList = calManager.Database.GetAll(Event.Uri, 0, 0);
+						calRecord = calList.GetCurrentRecord();
+						string summary = calRecord.Get<string>(Event.Summary);
+						int IsAllday = calRecord.Get<int>(Event.IsAllday);
+						CalendarTime sTime = Convert.ToBoolean(IsAllday)
+								? ConvertIntPtrToCalendarTime(new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc))
+								: calRecord.Get<CalendarTime>(Event.Start);
+						CalendarTime End = Convert.ToBoolean(IsAllday)
+										? ConvertIntPtrToCalendarTime(new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc))
+										: calRecord.Get<CalendarTime>(Event.End);
+						string location = calRecord.Get<string>(Event.Location);
+						string description = calRecord.Get<string>(Event.Description);
+						int Id = calRecord.Get<int>(Event.Id);
+						TimeSpan s = sTime.UtcTime - start;
+						TimeSpan e = end - End.UtcTime;
+						if (s.TotalSeconds >= 0 && e.TotalSeconds >= 0)
+						{
+							calendars.Add(
+								new CalendarEvent
+								{
+									Name = summary,
```

Comment:
Detabifying would still be greatly appreciated. This is a heck of a lot of indentation.